### PR TITLE
fix: Only show the `AccessibilityPage` if it is supported

### PR DESCRIPTION
**Pull request by spydon**

### CODE REVIEW

File: packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart [hunk 1]
```diff
@@ -13,7 +13,9 @@ class AccessibilityPage extends ConsumerWidget with ProvisioningPage {
   @override
   Future<bool> load(BuildContext context, WidgetRef ref) {
     final model = ref.read(accessibilityModelProvider);
-    return model.init().then((_) => true);
+    // This will return false when it is not Gnome, since the model wont be able
+    // to initialize with GSettings.
+    return model.init().then((_) => true, onError: (_) => false);
```
Comment (d-loose) [inline packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart:18]: Did you try this? Errors are caught and logged in `GnomeAccessibilityService`, so `model.init()` should never throw

Comment (spydon) [reply packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart:18]: I did not, I pushed it during the meeting 😄

Comment (d-loose) [reply packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart:18]: Just checked it by using nonexistent dconf paths and it doesn't work. Sorry, I think my `GnomeAccessibilityService` implementation fails a bit too gracefully in that case and just replies with default values. Maybe we could throw a custom exception in the `_tryGet`/`_trySet` methods

Comment (spydon) [reply packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart]: What about [this](https://github.com/canonical/ubuntu-desktop-provision/pull/406/commits/67a91f6c18988159f7a240e5f3aafffd364cf20c), if we don't get any options at all back, then we treat it as non-supported?

Comment (d-loose) [reply packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart]: works for me :+1:

**Review by d-loose — APPROVED** [review]

File: packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart (hunk 1 above)
Comment (spydon) [reply packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart]: Ah, if there are default values I guess that won't work either...

Comment (spydon) [reply packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart]: Seems to work!

Comment (spydon) [reply packages/ubuntu_provision/lib/src/accessibility/accessibility_page.dart]: Ah, haha, your comment didn't pop up until I commented

**Review by d-loose — CHANGES_REQUESTED** [review]
Wait sorry again :D the default values are an issue - I thought you had implemented this logic in `_tryGet`

File: packages/ubuntu_bootstrap/lib/services/gnome_accessibility_service.dart [hunk 2]
```diff
@@ -55,6 +55,10 @@ class GnomeAccessibilityService implements AccessibilityService {
     }
   }
 
+  @override
+  Future<bool> isSupported() =>
+      _a11yInterfaceSettings.list().then((_) => true, onError: (_) => false);
```
Comment (d-loose) [inline packages/ubuntu_bootstrap/lib/services/gnome_accessibility_service.dart:60]: In principle we should probably check all gsettings interfaces, but that should do for now :)

**Review by d-loose — APPROVED** [review]

**Pull request merged**